flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManagerDriver.java

```diff
@@ -360,7 +364,12 @@ public void onError(List<KubernetesPod> pods) {
 
         @Override
         public void handleFatalError(Throwable throwable) {
-            getResourceEventHandler().onError(throwable);
+            if (throwable instanceof KubernetesTooOldResourceVersionException) {
+                log.info("Creating a new watch on TaskManager pods.");
+                podsWatchOpt = watchTaskManagerPods();
```

Comment:
I think this should happen in the main thread context because we are modifying the state of the `KubernetesResourceMnaagerDriver`.